Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb

```diff
@@ -141,6 +141,392 @@
+       "      <td>0.019575</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5</th>\n",
+       "      <td>China</td>\n",
+       "      <td>6.136681</td>\n",
+       "      <td>813.418913</td>\n",
+       "      <td>0.007544</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                    Country  Life Growth (%)  GDP Growth (%)     Ratio\n",
+       "0                  Zimbabwe        31.956522      143.718552  0.222355\n",
+       "1                   Germany         3.846154       73.112644  0.052606\n",
+       "2  United States of America         3.255208       75.728155  0.042985\n",
+       "3                    Mexico         2.540107       68.545801  0.037057\n",
+       "4                     Chile         4.139715      211.475850  0.019575\n",
+       "5                     China         6.136681      813.418913  0.007544"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Calculate the growth percentage using the first year as the base in relation to the last year and the ratio.\n",
+    "def calculate_growth(df):\n",
+    "    df.sort_values('Year', inplace=True)\n",
+    "    initial_life_expectancy = df['Life'].iloc[0]\n",
+    "    initial_GDP = df['GDP'].iloc[0]\n",
+    "    df['Life Growth (%)'] = (df['Life'] / initial_life_expectancy - 1) * 100\n",
+    "    df['GDP Growth (%)'] = (df['GDP'] / initial_GDP - 1) * 100\n",
+    "    df['Ratio'] = df['Life Growth (%)'] / df['GDP Growth (%)'].replace({0:1})\n",
+    "    return df\n",
+    "\n",
+    "# group by country and apply the function.\n",
+    "growth_data = gdp_data.groupby('Country', group_keys=False).apply(calculate_growth)\n",
+    "\n",
+    "# calculate cumulative growth.\n",
+    "final_growth_data = growth_data.groupby('Country', group_keys=False).nth(-1).reset_index()\n",
+    "\n",
+    "# filter the columns\n",
+    "final_growth_percentages = final_growth_data[['Country', 'Life Growth (%)', 'GDP Growth (%)', 'Ratio']]\n",
+    "\n",
+    "# print results.\n",
+    "final_growth_percentages.sort_values('Ratio', ascending=False).reset_index(drop=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Analysis by countries\n",
+    "\n",
+    "Zimbabwe\n",
+    "\n",
+    "It shows the highest growth in life expectancy (approximately 31.96%) with a also significant GDP growth (approximately 143.72%). It has the highest ratio (0.222), suggesting that the improvement in life expectancy has been considerable even in the face of its economic growth. This is consistent with studies that claim that in countries with low development, variations in GDP have a greater impact on life expectancy because it directly affects literacy and nutrition of the population, which are two essential aspects in disease prevention impacting mortality.\n",
```

Comment:
Good analysis and interpretation. Check your sentences for some typos, and it'll be even better.

Reply:
Thanks, We will improve it.